### ISSUE TITLE
Possible issue with new readinto functionality

### ISSUE THREAD

**Issue by FoamyGuy** [adafruit/Adafruit_CircuitPython_ImageLoad#47]
I'm getting this error raised from the simpletest script in this repo:
```
Traceback (most recent call last):
  File "code.py", line 8, in <module>
  File "adafruit_imageload/__init__.py", line 57, in load
  File "adafruit_imageload/__init__.py", line 48, in load
  File "adafruit_imageload/bmp/__init__.py", line 59, in load
  File "adafruit_imageload/bmp/indexed.py", line 91, in load
TypeError: extra keyword arguments given
```
 My device is:
 ```
Adafruit CircuitPython 6.2.0-beta.4 on 2021-03-18; Adafruit Feather RP2040 with rp2040
```

**Comment by FoamyGuy** [comment]
I think we do have possibly two issues actually. 

1) It seems that the core release version made it to an "in-between" state where it does have `bitmaptools.read_into` but it's missing one of the arguments. This is causing the error mentioned above. I built from the current main branch and this error is not raised on that branch:
```
Adafruit CircuitPython 6.2.0-beta.4-53-ge084a9267-dirty on 2021-03-20; Adafruit Feather RP2040 with rp2040
```

2) there may also be some issue loading images on the branches that do have this argument.
Using the current release version of this library with `read_into()` I get this:
![image](https://user-images.githubusercontent.com/2406189/111890329-72eb2b80-89b6-11eb-8f55-b79da393ed53.png)


using imageload edited not to use `read_into()`
![image](https://user-images.githubusercontent.com/2406189/111890276-e0e32300-89b5-11eb-845d-9a1b7bdd38c4.png)


The image in use is here: https://github.com/adafruit/Adafruit_CircuitPython_ImageLoad/blob/master/examples/images/4bit.bmp

**Comment by FoamyGuy** [comment]
Interestingly some images are okay:
![image](https://user-images.githubusercontent.com/2406189/111890707-1ab62880-89ba-11eb-8dac-b548ce13e008.png)
this is one of the icons from the touch deck project. These are the ones I tested the changes with originally.

perhaps there is something specific about the `4bit.bmp` image causing trouble. I'll try out a few other images to see if any other have similar things going on.

@kmatch98 when you have a chance to take a look, I'm curious to see if you have ideas about what could be causing the differences in final image shown for this one.

**Comment by kmatch98** [comment]
@FoamyGuy I confirmed your issue.  I was able to adjust the inputs to `_bitmap_readinto` to get the shapes to be correct, but some of the colors displayed are all wrong.

I verified that the loaded palette colors exactly the same with both versions.  Perhaps the pixel index reads are off by a few bits somewhere?  I'm not sure I'll have any time tomorrow to work on this, and anyway we may need to bring jepler into consult.


I added the `swap_bytes_in_element = False` as shown below and it corrected the shape, but not the colors.
```python
            if _bitmap_readinto:
                _bitmap_readinto(
                    bitmap,
                    file,
                    bits_per_pixel=color_depth,
                    element_size=4,
                    reverse_pixels_in_element=True,
                    swap_bytes_in_element=False,
                    reverse_rows=True,
                )
```

**Comment by kmatch98** [comment]
Just a little bit more information to confirm that some of the bitmap color indexes are off by 8 when using the new `readinto` function.  Interestingly, only certain columns or rows seem to be affected.

Here are the bitmap color indexes when using the previous "standard speed" version:
```
5 5 5 11 11 6 6 6 14 14 14 3 3 4 4 4 5  
5 5 5 11 11 6 6 6 14 14 14 3 3 4 4 4 5  
5 5 5 11 11 6 6 6 14 14 14 3 3 4 4 4 5  
5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5  
5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5  
5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5  
5 5 5 15 15 1 1 1 9 9 9 0 0 12 12 12 5  
5 5 5 15 15 1 1 1 9 9 9 0 0 12 12 12 5  
5 5 5 15 15 1 1 1 9 9 9 0 0 12 12 12 5  
5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5  
5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5  
5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5  
5 5 5 2 2 10 10 10 8 8 8 13 13 7 7 7 5  
5 5 5 2 2 10 10 10 8 8 8 13 13 7 7 7 5  
5 5 5 2 2 10 10 10 8 8 8 13 13 7 7 7 5  
```

Here are the bitmap color index values with the new "fast" `readinto` version:
```
5 5 5 3 3 6 6 6 6 6 6 3 3 4 4 4 5  
5 5 5 3 3 6 6 6 6 6 6 3 3 4 4 4 5  
5 5 5 3 3 6 6 6 6 6 6 3 3 4 4 4 5  
5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5  
5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5  
5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5  
5 5 5 7 7 1 1 1 1 1 1 0 0 4 4 4 5  
5 5 5 7 7 1 1 1 1 1 1 0 0 4 4 4 5  
5 5 5 7 7 1 1 1 1 1 1 0 0 4 4 4 5  
5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5  
5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5  
5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5  
5 5 5 2 2 2 2 2 0 0 0 5 5 7 7 7 5  
5 5 5 2 2 2 2 2 0 0 0 5 5 7 7 7 5  
5 5 5 2 2 2 2 2 0 0 0 5 5 7 7 7 5 
```

**Comment by kmatch98** [comment]
Ok, I found the issue, was an incorrect bitmask for BMPs with 4-bit indexes, along with the `reverse_pixels_in_element=True`.

This will require:
 - update to `bitmaptools.readinto`, PR is here: https://github.com/adafruit/circuitpython/pull/4455
 - update to this library, PR is here: https://github.com/adafruit/Adafruit_CircuitPython_ImageLoad/pull/49

**Comment by kmatch98** [comment]
I think the two things above were merged so all should be good to close out this issue.

**Comment by FoamyGuy** [comment]
This was resolved by #48 , #49 , and [#4454](https://github.com/adafruit/circuitpython/pull/4454) in the core.
